pegasus/test/test_pegasus_documents.rb

```diff
@@ -0,0 +1,186 @@
+require_relative './test_helper'
+require_relative '../router'
+require 'helpers/auth_helpers'
+require 'cdo/rack/request'
+require 'parallel'
+require 'open3'
+require 'digest'
+require 'yaml'
+require 'tempfile'
+require 'zlib'
+
+# Runs integration tests against the full Pegasus document repository.
+class PegasusTest < Minitest::Test
+  include Rack::Test::Methods
+
+  def app
+    Documents.new
+  end
+
+  def test_pegasus_documents
+    documents = Documents.new.helpers.all_documents.map do |page|
+      "#{page[:site]}#{page[:uri]}"
+    end
+    CDO.log.info "Found #{documents.length} Pegasus documents."
+    assert_operator documents.length, :>, 2000
+  end
+
+  # All documents expected to return 200 status-codes, with the following exceptions:
+  STATUS_EXCEPTIONS = {
+    302 => %w[
+      code.org/educate
+      code.org/teacher-dashboard
+      code.org/teacher-dashboard/review-hociyskvuwa
+      code.org/teach code.org/student
+    ],
+    301 => %w[
+      csedweek.org/resource_kit
+    ],
+    401 => %w[
+      code.org/create-company-profile
+    ]
+  }
+
+  # All documents expected to return 'text/html' content-type, with the following exceptions:
+  CONTENT_TYPE_EXCEPTIONS = {
+    'text/plain' => %w[
+      code.org/health_check
+      code.org/robots.txt
+    ],
+    'text/ng-template' => %w[
+      code.org/teacher-dashboard/section_projects
+      code.org/teacher-dashboard/section_stats
+      code.org/teacher-dashboard/section
+      code.org/teacher-dashboard/section_progress
+      code.org/teacher-dashboard/sections
+      code.org/teacher-dashboard/section_assessments
+      code.org/teacher-dashboard/nav
+      code.org/teacher-dashboard/landing
+      code.org/teacher-dashboard/plan
+      code.org/teacher-dashboard/student
+      code.org/teacher-dashboard/section_responses
+      code.org/teacher-dashboard/signin_cards
+      code.org/teacher-dashboard/section_manage
+    ]
+  }
+
+  # All documents expected to have unchanged content between runs, with the following exceptions:
+  # (TODO: remove random elements from these pages.)
```

Comment:
Can you expand on what you mean by random elements?

Reply:
From the [PR description comment](https://github.com/code-dot-org/code-dot-org/pull/21143#issue-173901654):

> Some documents (marked as hard-coded exceptions) render content that changes on each run (e.g., there's some randomness in the HTML response). Considering most Pegasus pages are CDN-cached causing random page contents to be unavailable in practice, I consider these random-content exceptions to be a form of technical-debt that should be cleaned up at some point.

Reply:
Ah sorry - when I read random, I inferred elements not tested, but you actually mean random. This makes sense, thanks.